pay-queue/src/pay_queue/services/eft/eft_reconciliation.py

```diff
@@ -24,14 +29,70 @@
 from pay_api.services.eft_short_name_historical import EFTShortnameHistorical as EFTHistoryService
 from pay_api.services.eft_short_name_historical import EFTShortnameHistory as EFTHistory
 from pay_api.services.eft_short_names import EFTShortnames as EFTShortnamesService
+from pay_api.services.email_service import send_email as send_email_service
 from pay_api.utils.enums import EFTFileLineType, EFTPaymentActions, EFTProcessStatus
 from sentry_sdk import capture_message
 
+from pay_queue.auth import get_token
 from pay_queue.minio import get_object
 from pay_queue.services.eft import EFTHeader, EFTRecord, EFTTrailer
 
 
-def reconcile_eft_payments(msg: Dict[str, any]):  # pylint: disable=too-many-locals
+class EFTReconciliationContext:
+    """Initialize the EFTReconciliationContext."""
+
+    def __init__(self, ce):
+        """:param ce: The cloud event object containing relevant data."""
+        self.ce = ce
+        self.msg = ce.data
+        self.file_name: str = self.msg.get('fileName')
+        self.minio_location: str = self.msg.get('location')
+        self.error_messages: List[Dict[str, any]] = []
+
+    def eft_error_handling(self, row, error_msg: str, ex: Exception = None,  # pylint:disable=too-many-arguments
+                           capture_error: bool = True, send_email: bool = False, table_name: str = None):
+        """Handle EFT errors by logging, capturing messages, and optionally sending an email."""
+        if ex:
+            formatted_traceback = ''.join(traceback.TracebackException.from_exception(ex).format())
+            error_msg = f'{error_msg}\n{formatted_traceback}'
+        if capture_error:
+            current_app.logger.error(error_msg)
+            capture_message(error_msg, level='error')
+        self.error_messages.append({'error': error_msg, 'row': row})
+        if send_email:
+            self.send_error_email(self.error_messages, table_name)
+
+    def send_error_email(self, error_messages, table_name):
```

Comment:
Is there a need for an email service? I'm thinking yes?

Reply:
Move this into it as well

![image](https://github.com/user-attachments/assets/c614ff83-eab5-46c6-9347-ceb057522cfb)